### ISSUE TITLE
Proposal: Improve Notification colour contrast (a11y)

### ISSUE THREAD

**Issue by m10l** [contentful/forma-36#446]
## The problem

The Notification component currently fail WCAG colour contrast accessibility guidelines:

**Notification with `success` intent:**
> Element has insufficient color contrast of 2.99 (foreground color: #ffffff, background color: #0baa75, font size: 10.5pt (14px), font weight: bold). Expected contrast ratio of 4.5:1

**Notification with `warning` intent:** 
> Element has insufficient color contrast of 2.85 (foreground color: #ffffff, background color: #db8500, font size: 10.5pt (14px), font weight: bold). Expected contrast ratio of 4.5:1

## The proposed solution

I'd like to propose some changes to these components to make sure we can consistently meet accessibility guidelines by not setting text on a solid-colour background.

Here are some mockups of what I'd like to implement:

**Success**
<img width="510" alt="Screenshot 2020-03-30 at 15 03 25" src="https://user-images.githubusercontent.com/4446634/77915740-42d3e580-7298-11ea-8c91-44672dcc6ef5.png">

**Warning**
<img width="513" alt="Screenshot 2020-03-30 at 15 01 35" src="https://user-images.githubusercontent.com/4446634/77915750-48313000-7298-11ea-8096-6e207d9603be.png">

**Error**
<img width="511" alt="Screenshot 2020-03-30 at 15 01 17" src="https://user-images.githubusercontent.com/4446634/77915756-4cf5e400-7298-11ea-9746-deaa7ef25c6a.png">

**Comment by gui-santos** [comment]
IMO, they look great! It really improves the readability of the notifications 👍

**Comment by domarku** [comment]
Great proposal @m10l ✨ I've got couple of comments:
- This definitely makes them more readable!
- There should be some more vertical spacing between the text elements, `spacing-2xs` (4px) or even `xs` (8px), to make the message easier to scan and the perceived click area around the CTA larger
- The icon should be horizontally aligned with the Title / Close button. People scan messages in the form of the letter F, but this way the eye pops back left to view the icon after reading the Title. Examples from other systems: [Atlassian](https://atlassian.design/guidelines/product/components/flag-messages), [Polaris](https://polaris.shopify.com/components/feedback-indicators/banner#navigation), [Carbon](https://www.carbondesignsystem.com/components/notification/code/)
- Can we use the [TextLink](https://forma-36-react-components.netlify.com/?path=/story/components-textlink--default) component for the CTA here? It would make it stand out a bit more and consistent with the rest of the web app.
- Consider text wrapping in the body. The right side of the notification — close button with white background — should have similar spacing as the left side — icon with coloured background, so no text goes below the close button.
- Where is the CTA when there's no body text? An example might be a "success" message with an Undo
- Could we rename the component to Message or Toast or would that be a breaking change?

Here's a sketch of some details I mentioned above:
![Group](https://user-images.githubusercontent.com/7631029/77927518-9c8fdc00-72a7-11ea-8add-af130ddaa58e.png)

**Comment by domarku** [comment]
I see that Github has the icon vertically centered in their [Toast component](https://primer.style/css/components/toasts). Look at others to get inspired not to prove a point! :)

**Comment by m10l** [comment]
@domarku Awesome, thanks for the review and suggestions

> There should be some more vertical spacing between the text elements, spacing-2xs (4px) or even xs (8px), to make the message easier to scan and the perceived click area around the CTA larger

Cool, I think this is worth tackling at the same time too so will do this as part of the same PR

> The icon should be horizontally aligned with the Title / Close button. People scan messages in the form of the letter F, but this way the eye pops back left to view the icon after reading the Title. Examples from other systems: Atlassian, Polaris, Carbon

The examples linked all have a solid background for the entire notification, so I think your suggestion makes sense there, but doesn't work for what we're doing as it's a smaller section with a solid colour background.

> Can we use the TextLink component for the CTA here? It would make it stand out a bit more and consistent with the rest of the web app.

Ah good spot, was using TextLink but the wrong colour.

> Consider text wrapping in the body. The right side of the notification — close button with white background — should have similar spacing as the left side — icon with coloured background, so no text goes below the close button.

Cool, think text is already prevented from going beneath the close button, but I'll test it to make sure.

> Where is the CTA when there's no body text? An example might be a "success" message with an Undo

It's always at the bottom

Could we rename the component to Message or Toast or would that be a breaking change?

> That'd be a breaking change, wouldn't recommend it for now

> Should we replace the icon in the red error message with the Error Icon to make it different than the icon for the warning message?

Ooo, definitely, good shout.

**Comment by domarku** [comment]
Right on @m10l! Thanks for addressing all my comments. Any chance you would consider putting the CTA inline in case of no body text? Like so:
![Group Copy 8](https://user-images.githubusercontent.com/7631029/77936943-e54d9200-72b3-11ea-96bd-2feeb8ce5985.png)

**Comment by m10l** [comment]
> Right on @m10l! Thanks for addressing all my comments. Any chance you would consider putting the CTA inline in case of no body text? Like so:
> ![Group Copy 8](https://user-images.githubusercontent.com/7631029/77936943-e54d9200-72b3-11ea-96bd-2feeb8ce5985.png)

@domarku Still not in favour of doing this, I think introduces more potential problems than it tries to solve.

**Pros/cons**
- ➕ Everything readable on a single line
- ➖Concerns around truncation - how to handle if the CTA and title are both super long? What if the title spans multiple lines?
- ➖Consistency - I'd argue that moving the CTA based on number of lines of text could create a jarring experience for the user

**Comment by domarku** [comment]
Ah, good points. Although that makes me think we should have come copy guidelines for the notifications. I'll give it a go at writing and adding them to the thread later in the day.

**Comment by m10l** [comment]
@domarku That'd be awesome! Shall I close this an open a PR?

**Comment by domarku** [comment]
Do it 🤠

**Comment by wichniowski** [comment]
looks really good! now if you compare the old notification to it, the old one look super outdated and wrong so 100x for the proposal

**Comment by m10l** [comment]
Closing as PR created here: https://github.com/contentful/forma-36/pull/449
